mountpoint-s3/src/checksums.rs

```diff
@@ -275,26 +316,69 @@ mod tests {
         let original = ChecksummedBytes::new(bytes, checksum);
         let slice = original.slice(range);
 
-        assert_eq!(expected, original.orig_bytes);
-        assert_eq!(expected, original.curr_slice);
-        assert_eq!(expected, slice.orig_bytes);
-        assert_eq!(expected_slice, slice.curr_slice);
+        assert_eq!(expected, original.buffer);
+        assert_eq!(expected, original.buffer_slice());
+        assert_eq!(expected, slice.buffer);
+        assert_eq!(expected_slice, slice.buffer_slice());
         assert_eq!(checksum, original.checksum);
         assert_eq!(checksum, slice.checksum);
     }
 
+    fn slice(original: Range<usize>, range: impl RangeBounds<usize>) -> ChecksummedBytes {
+        let buffer = Bytes::copy_from_slice(&vec![0; original.len()]);
+        let checksum = crc32c::checksum(&buffer);
+        let bytes = ChecksummedBytes {
+            buffer,
+            range: original,
+            checksum,
+        };
+        bytes.slice(range)
+    }
```

Comment:
I'd prefer we just include this setup inline in the test. We can simplify the code a bit too:

```rust
let buffer = Bytes::from_static([0; original.len()]);
let bytes = ChecksummedBytes::from_bytes(buffer);
let slice = bytes.slice(range);
```

Reply:
better yet, we could just add (maybe test only) `ChecksummedBytes::from_static`? test setup / 'boilerplate' becomes:

```rust
let bytes = ChecksummedBytes::from_static([0; original.len()]);
let slice = bytes.slice(range);
```

Reply:
Your suggestion is missing the `original` range, but you are definitely right about the `slice` helper function not being clear! 
In fact, I think the issue is that it is both setting up the instance to test and invoking the `slice` method on it. I've fixed that by using the helper function just for the setup and calling `slice` in each test function. I hope that it's clearer now.

Reply:
thanks, yep clearer!